scripts/s3-dist.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -e
+
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# This script should be run from the repo's root directory (same directory as serverless.yaml)
+# cd fhir-works-on-aws-deployment
+# chmod +x ./scripts/s3-dist.sh && ./scripts/s3-dist.sh <BUCKET_NAME> <BUCKET_REGION>
+
+# Package output directory is assumed to be default (.serverless). Passing -p flag to `sls package` can break this script.
+# Dependencies: npm, aws CLI (configured), sed, jq
+if [ -z "$1" ] || [ -z "$2" ]; then
+    echo Please provide both the prefix and region for the bucket
+    echo For example: "$0" solutions us-east-1
+    exit 1
+fi
+
+# Bucket name is assumed to be $BUCKET_PREFIX-$BUCKET_REGION (e.g. solutions-us-east-1)
+BUCKET_PREFIX=$1
+BUCKET_REGION=$2
+
+function cleanup(){
+    echo Removing deploymentBucket from serverless.yaml
+    sed -i -e '20d' serverless.yaml
```

Comment:
What is this doing removing line 20?

Reply:
[Line 33](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/88c9574338f48dc0a7241de5e5f6856061335cac/scripts/s3-dist.sh#L33) adds provider.deploymentBucket to line 20. This prevents `sls package` from creating a default bucket, and using the parameters provided to the script instead. When the script exits, it removes it here (trapping the EXIT signal on [line 34](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/88c9574338f48dc0a7241de5e5f6856061335cac/scripts/s3-dist.sh#L34)), incase developers are already using the default bucket. 

It's not very portable since any changes that may move line 20 outside of the `provider` segment of the serverless.yaml will break this. I couldn't think of a better way for the Solutions team to provide their own bucket. Open to other suggestions.

Reply:
Check this out I think this is what we want? https://medium.com/serverlessguru/how-to-reuse-an-aws-s3-bucket-for-multiple-serverless-framework-deployments-d1673d3d8259

Reply:
This blog is what lead me to this solution actually, using provider.deploymentBucket to use an existing bucket. This way the Solutions team can provide their own existing buckets (solutions-<region-name>) at build time (when they run the script) so customers can deploy the template into their own accounts. Without this somewhat hacky automation, the Solutions team would have to provide the deploymentBucket parameter to the serverless.yaml template themselves. Or was there something else in the blog that I missed?

Reply:
> Without this somewhat hacky automation, the Solutions team would have to provide the deploymentBucket parameter to the serverless.yaml template themselves.

What is wrong with that? We could create a new serverless CLI variable 'deployBucket' and pass that in?

`sls package --region "$BUCKET_REGION" --deployBucket '"$BUCKET_PREFIX"'-'"$BUCKET_REGION"'`

https://www.serverless.com/framework/docs/providers/aws/guide/variables/#referencing-cli-options

Reply:
Passing --deployBucket means provider.deploymentBucket already exists in the template, right? This would work, as long as developers that will be running sls commands are okay with always providing a deploymentBucket. Or is there an sls native way to ignore provider.deploymentBucket if --deployBucket isn't provided?

Reply:
> deployBucket means provider.deploymentBucket already exists in the template, right?
Actually it is worse it means that the bucket exists in the account.

hmmm ok so it sounds like this is the best approach. I think at the very least we need to remove the hard coded 19/20 line number. Could you look into a way to get the line number `providers:` is on?

Maybe this is a good lead: https://stackoverflow.com/a/18870500/9041033


Similar problem: https://forum.serverless.com/t/how-to-create-deploymentbucket-when-manually-specified-if-it-doesnt-exist/2439